src/ess/reflectometry/workflow.py

```diff
@@ -0,0 +1,67 @@
+from collections.abc import Hashable, Sequence
+from itertools import chain
+
+import pandas as pd
+import sciline
+import scipp as sc
+
+from ess.amor.types import RawChopper
+from ess.reflectometry.orso import (
+    OrsoExperiment,
+    OrsoOwner,
+    OrsoSample,
+    OrsoSampleFilenames,
+)
+from ess.reflectometry.types import (
+    Filename,
+    FootprintCorrectedData,
+    RunType,
+    SampleRotation,
+    SampleRun,
+)
+
+
+def _set_first_if_lost(coord, da, das):
+    if coord not in da.coords:
+        da.coords[coord] = das[0].coords[coord]
+
+
+def _concatenate_event_lists(*das):
+    da = sc.reduce(das).bins.concat()
+    _set_first_if_lost('position', da, das)
+    _set_first_if_lost('sample_rotation', da, das)
+    _set_first_if_lost('detector_rotation', da, das)
+    return da
```

Comment:
```suggestion
def _concatenate_event_lists(*das):
    return sc.reduce(das).bins.concat().assign_coords(das[0].coords)
```
but shouldn't we check that they are at lost close?

Reply:
> but shouldn't we check that they are at lost close?

No I don't think we can do that in any reasonable way. What is "close" depends on  the context.

In this code we should just expect that what we're given is reasonable, and then we can add a validation layer at a higher level that takes the context into account.

Reply:
I'll use this suggestion :+1:
with the minor modification that I'll only propagate the coords that are explicitly listed,
to avoid propagating something that actually should be dropped.